### PR TITLE
API: Add content-type to all structured error responses

### LINKED ISSUE #2596: Missing content-type application/json;charset=utf-8 for status code 404 on /wallets
# Context

I'm building a library for Elixir that talks to the cardano-wallet and I think I found a bug.

The content-type header seems to be missing when calling http://localhost:8090/v2/wallets/<wallet_id> with an non existing id which returns with status code 404. 

This is the headers I get back:
    {"date", "Thu, 01 Apr 2021 21:12:48 GMT"},
    {"server", "Warp/3.3.5"},
    {"content-length", "122"}

If I call the same endpoint but with a badly formatted id which returns a status code 400 I get these headers:
    {"date", "Thu, 01 Apr 2021 21:12:48 GMT"},
    {"server", "Warp/3.3.5"},
    {"content-length", "92"},
    {"content-type", "application/json;charset=utf-8"}

The problem with this is that the content type is used by many http clients to automatically decode the json response to a struct so the implementation will be cluttered with workarounds.

## Essential Information

 - **Version**:      master
 - **Platform**:     Linux
 - **Installation**: Built from source
 - **Network**: testnet

### PR DESCRIPTION
**Pull request by rvl**
### Issue Number

#2596 / ADP-853

### Overview

- [x] Make the `LiftHandler` type class simpler.
- [x] Add content-type header for all uses of `apiError`
- [x] Add regression test.
- [x] Add content-type to generic api error handler tests.

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/Api/Server.hs [hunk 1]
```diff
@@ -2434,23 +2437,28 @@ withWorkerCtx ctx wid onMissing onNotResponding action =
...
 
 apiError :: ServerError -> ApiErrorCode -> Text -> ServerError
 apiError err code message = err
     { errBody = Aeson.encode $ Aeson.object
         [ "code" .= code
         , "message" .= T.replace "\n" " " message
         ]
+    , errHeaders =
+        (hContentType, renderHeader $ contentType $ Proxy @JSON)
+        : errHeaders err
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/Api/Server.hs:2461]: :+1:

File: lib/core/test/unit/Cardano/Wallet/Api/ServerSpec.hs [hunk 2]
```diff
@@ -214,3 +218,60 @@ spec = describe "API Server" $ do
...
+    it "ErrNoSuchWallet" $ do
+        let handler :: ExceptT ErrNoSuchWallet IO ()
+            handler = throwE $ ErrNoSuchWallet wid
+            wid = unsafeFromText "0000000000000000000000000000000000000000"
+        let expectedErr = ServerError
+                { errHTTPCode = 404
+                , errReasonPhrase = "Not Found"
+                , errBody = mconcat
+                    [ "{\"code\":\"no_such_wallet\",\"message\":\"I couldn't "
+                    , "find a wallet with the given id: 0000000000000000000000000000000000000000\"}" ]
```
Comment (KtorZ) [inline lib/core/test/unit/Cardano/Wallet/Api/ServerSpec.hs]: We could avoid the message here, this is tested in other layers and will only be painful to maintain over time.

File: lib/core/test/unit/Cardano/Wallet/ApiSpec.hs [hunk 3]
```diff
@@ -169,41 +173,40 @@ spec = parallel $ do
...
+assertErrorResponse
+    :: HasCallStack
+    => Int -- ^ Expected status
+    -> Text -- ^ Expected error code string
+    -> ExpectedError
+    -> SResponse
+    -> Session ()
+assertErrorResponse status code (ExpectedError msg) response = do
+    response & assertStatus status
+    response & assertHeader "Content-Type" "application/json;charset=utf-8"
```
Comment (KtorZ) [inline lib/core/test/unit/Cardano/Wallet/ApiSpec.hs:185]: :clap:

**Review by KtorZ — APPROVED** [review]

File: lib/core/test/unit/Cardano/Wallet/Api/ServerSpec.hs (hunk 2 above)
Comment (rvl) [reply lib/core/test/unit/Cardano/Wallet/Api/ServerSpec.hs]: OK I changed the assertions to be more lenient and maintainable.
As far as I can see, the error message is tested by integration tests but nowhere else.

File: lib/core/test/unit/Cardano/Wallet/ApiSpec.hs (hunk 3 above)
Comment (rvl) [reply lib/core/test/unit/Cardano/Wallet/ApiSpec.hs:185]: These tests don't actually cover the `no_such_wallet` error reported in issue #2596.

**Comment by rvl** [comment]
Thanks for the review comments.

bors r+

File: lib/core/test/unit/Cardano/Wallet/ApiSpec.hs (hunk 3 above)
Comment (KtorZ) [reply lib/core/test/unit/Cardano/Wallet/ApiSpec.hs:185]: Yes indeed, the `404` are sort of cumbersome to trigger with this generic approach which only focuses on what exists.

**Pull request merged**